src/Scripting/Core/ObjectFormatter.cs

```diff
@@ -82,70 +95,68 @@ private static DebuggerTypeProxyAttribute GetApplicableDebuggerTypeProxyAttribut
             return null;
         }
 
-        private static bool IsApplicableAttribute(Type type, Type targetType, string targetTypeName)
+        private static bool IsApplicableAttribute(TypeInfo type, TypeInfo targetType, string targetTypeName)
         {
-            return type != null && targetType.IsEquivalentTo(type)
+            return type != null && AreEquivalent(targetType, type)
                 || targetTypeName != null && type.FullName == targetTypeName;
         }
 
+        private static bool AreEquivalent(TypeInfo type, TypeInfo other)
+        {
+            // TODO: does Equals unify NoPIA interfaces?
+            return type.Equals(other);
+        }
+
         private static object GetDebuggerTypeProxy(object obj)
         {
             // use proxy type if defined:
-            Type type = obj.GetType();
+            var type = obj.GetType().GetTypeInfo();
             var debuggerTypeProxy = GetApplicableDebuggerTypeProxyAttribute(type);
             if (debuggerTypeProxy != null)
             {
-                var proxyType = Type.GetType(debuggerTypeProxy.ProxyTypeName, false, false);
-                if (proxyType != null)
+                try
                 {
-                    try
+                    var proxyType = Type.GetType(debuggerTypeProxy.ProxyTypeName, throwOnError: false, ignoreCase: false);
+                    if (proxyType != null)
                     {
-                        if (proxyType.IsGenericTypeDefinition)
+                        if (proxyType.GetTypeInfo().IsGenericTypeDefinition)
                         {
-                            proxyType = proxyType.MakeGenericType(type.GetGenericArguments());
+                            proxyType = proxyType.MakeGenericType(type.GenericTypeArguments);
                         }
 
-                        return Activator.CreateInstance(
-                            proxyType,
-                            Ref.BindingFlags.Instance | Ref.BindingFlags.NonPublic | Ref.BindingFlags.Public,
-                            null,
-                            new object[] { obj },
-                            null,
-                            null
-                        );
-                    }
-                    catch (Exception)
-                    {
-                        // no-op, ignore proxy if it is implemented incorrectly
+                        return Activator.CreateInstance(proxyType, new object[] { obj });
                     }
                 }
+                catch (Exception)
+                {
+                    // no-op, ignore proxy if it is implemented incorrectly or can't be loaded
+                }
             }
 
             return null;
         }
 
-        private static Ref.MemberInfo ResolveMember(object obj, string memberName, bool callableOnly)
+        private static MemberInfo ResolveMember(object obj, string memberName, bool callableOnly)
```

Comment:
If 'callableOnly' is false, do we exclude methods? If so, consider renaming, perhaps to "isCallable".